### PR TITLE
Matching View (Mentor-Mentee matching)

### PR DESCRIPTION
**Pull request by sleidig**

### CODE REVIEW

File: src/app/features/matching-entities/matching-entities/matching-entities.component.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      );
+    } else {
+      await this.entityMapper.save(newMatchEntity);
+    }
+
+    // lock in current selection to avoid duplicate matches and provide user feedback
+    this.sideDetails.forEach((s) => (s.availableEntities = undefined));
+  }
+
+  applySelectedFilters(side: MatchingSide, $event: DataFilter<Entity>) {
```
Comment (TheSlimvReal) [inline src/app/features/matching-entities/matching-entities/matching-entities.component.ts:202]:
```suggestion
  applySelectedFilters(side: MatchingSide, filter: DataFilter<Entity>) {
```
Let's not make this ambiguous. I am very worried that `$event` might be something internal (like it is in the template)

File: src/app/utils/utils.ts [hunk 2]
```diff
@@ -132,6 +132,12 @@ export function parseJwt(token): {
   return JSON.parse(jsonPayload);
 }
 
+export function addAlphaToHexColor(color, opacity) {
+  // coerce values so ti is between 0 and 1.
+  var _opacity = Math.round(Math.min(Math.max(opacity || 1, 0), 1) * 255);
+  return color + _opacity.toString(16).toUpperCase();
```
Comment (TheSlimvReal) [inline src/app/utils/utils.ts:138]:
```suggestion
  // coerce values so it is between 0 and 1.
  const opacity = Math.round(Math.min(Math.max(opacity || 1, 0), 1) * 255);
  return color + opacity.toString(16).toUpperCase();
```

**Review by TheSlimvReal — CHANGES_REQUESTED** [review]
Great work, the UI already works very seamlessly and it also feels simple which I like. 
I think the code can be made a little more explicit in order to have less *string hacking*. But these things are only smaller refactorings.

- [ ] Clicking 'x' in the 'create matching' popup disables the component even though nothing has been saved, until the page is opened again. I think the component should only change, once the new entity has actually been saved.
- [ ] In the table there are no labels to the columns. For some properties it might be a bit unclear. E.g. in the example when a school is selected there is just a ✔️ without a label explaining what it means (you also have this as a TODO)

File: src/app/core/form-dialog/form-dialog.service.ts [hunk 3]
```diff
@@ -66,6 +79,51 @@ export class FormDialogService {
...
+    const columns: FormFieldConfig[] = this.inferFormFieldColumns(
+      columnsOverall,
+      entity
+    );
+
+    const columnsToDisplay = columns
+      .filter((col) => typeof col === "string" || col.edit)
+      .map((col) =>
+        typeof col === "string"
+          ? col
```
[the remark this answers is not in the document]
Comment (sleidig) [reply src/app/core/form-dialog/form-dialog.service.ts:93]: yes, you are right ... This is left from a previous attempt to make it work :sweat_smile:   thanks for pointing it out!

File: src/app/features/matching-entities/matching-entities/matching-entities.component.html [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    <tr mat-header-row *matHeaderRowDef="columnsToDisplay"></tr>
+    <tr mat-row *matRowDef="let row; columns: columnsToDisplay;"></tr>
+  </table>
+</div>
+
+<div class="margin-bottom-large flex-column gap-small">
+  <div *ngIf="showMap" class="flex-grow map-placeholder">Map</div>
+  <button mat-stroked-button color="accent" style="width: 100%"
+          (click)="createMatch()"
+          [disabled]="!sideDetails[0]?.selected || !sideDetails[1]?.selected || (!sideDetails[0].availableEntities && !sideDetails[1].availableEntities)">
```
[the remark this answers is not in the document]
Comment (sleidig) [reply src/app/features/matching-entities/matching-entities/matching-entities.component.html:44]: ok, fair point to make this more explicit with its own flag for "already matched" :+1:

File: src/app/features/matching-entities/matching-entities.module.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    CommonModule,
+    MatCardModule,
+    EntitySubrecordModule,
+    MatButtonModule,
+    ViewModule,
+    MatTableModule,
+    MatTooltipModule,
+    FilterModule,
+  ],
+  exports: [MatchingEntitiesComponent],
```
[the remark this answers is not in the document]
Comment (sleidig) [reply src/app/features/matching-entities/matching-entities.module.ts]: actually needed for easier integration in Storybook - and maybe fair to export it if it is also available as dynmic component?

File: src/app/features/matching-entities/matching-entities/matching-entities.component.html [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                             matTooltip="These details of the record are displayed to make comparison easier. You cannot select a different record here."
+                             i18n-matTooltip="tooltip explaining that this entity in matching is locked"
+        ></app-fa-dynamic-icon>
+      </th>
+
+      <td mat-cell *matCellDef="let property">
+        <ng-container
+          *ngIf="side.selected?.[property[i]] && side.entityType.schema.get(property[i]); else placeholder"
+          [appDynamicComponent]="{
+          component: schemaService.getComponent(side.entityType.schema.get(property[i])),
```
[the remark this answers is not in the document]
Comment (sleidig) [reply src/app/features/matching-entities/matching-entities/matching-entities.component.html:19]: moving this into a onPush component that handles things in a reusable way.

**Review by TheSlimvReal — APPROVED** [review]
Works great now.

Generally, it always hurts me to see the coverage go down a bit but I can also see that the tests would not add too much value as much of the code is rather trivial. I leave it up to you.

**Pull request merged**